fdbserver/workloads/Mako.actor.cpp

```diff
@@ -0,0 +1,530 @@
+			// TODO: if there is any other traceEvent to add
+			TraceEvent((self->description() + "_CommitLatency").c_str()).detail("Mean", self->opLatencies[OP_COMMIT].mean()).detail("Median", self->opLatencies[OP_COMMIT].median()).detail("Percentile5", self->opLatencies[OP_COMMIT].percentile(.05)).detail("Percentile95", self->opLatencies[OP_COMMIT].percentile(.95)).detail("Count", self->opCounters[OP_COMMIT].getValue()).detail("Elapsed", elapsed);
+			TraceEvent((self->description() + "_GRVLatency").c_str()).detail("Mean", self->opLatencies[OP_GETREADVERSION].mean()).detail("Median", self->opLatencies[OP_GETREADVERSION].median()).detail("Percentile5", self->opLatencies[OP_GETREADVERSION].percentile(.05)).detail("Percentile95", self->opLatencies[OP_GETREADVERSION].percentile(.95)).detail("Count", self->opCounters[OP_GETREADVERSION].getValue());
+			
+			std::string ts = format("T=%04.0fs: ", elapsed);
+			self->periodicMetrics.push_back(PerfMetric(ts + "Transactions/sec", (self->xacts.getValue() - last_xacts) / self->periodicLoggingInterval, false));
+			self->periodicMetrics.push_back(PerfMetric(ts + "Operations/sec", (self->totalOps.getValue() - last_ops) / self->periodicLoggingInterval, false));
+
+			last_xacts = self->xacts.getValue();
+			last_ops = self->totalOps.getValue();
+		}
+	}
+	ACTOR Future<Void> _setup(Database cx, MakoWorkload* self) {
+
+		state Promise<double> loadTime;
+		state Promise<vector<pair<uint64_t, double>>> ratesAtKeyCounts;
+
+		wait(bulkSetup(cx, self, self->rowCount, loadTime, self->insertionCountsToMeasure.empty(), self->warmingDelay,
+		               self->maxInsertRate, self->insertionCountsToMeasure, ratesAtKeyCounts));
+
+		// This is the setup time 
+		self->loadTime = loadTime.getFuture().get();
+		// This is the rates of importing keys
+		self->ratesAtKeyCounts = ratesAtKeyCounts.getFuture().get();
+
+		return Void();
+	}
+
+	ACTOR Future<Void> _start(Database cx, MakoWorkload* self) {
+		vector<Future<Void>> clients;
+		for (int c = 0; c < self->actorCountPerClient; ++c) {
+			clients.push_back(self->makoClient(cx, self, self->actorCountPerClient / self->transactionsPerSecond, c));
+		}
+
+		if (self->enableLogging)
+			clients.push_back(tracePeriodically(self));
+
+		wait( timeout( waitForAll( clients ), self->testDuration, Void() ) );
+		return Void();
+	}
+
+	ACTOR Future<Void> makoClient(Database cx, MakoWorkload* self, double delay, int actorIndex) {
+
+		state Key rkey;
+		state Value rval;
+		state Transaction tr(cx);
+		state bool doCommit;
+		state int i, count;
+		state uint64_t range, indBegin, indEnd, rangeLen;
+		state double lastTime = now();
+		state double commitStart;
+		state KeyRangeRef rkeyRangeRef;
+		state vector<int> perOpCount(MAX_OP, 0);
+
+		TraceEvent("ClientStarting").detail("AcotrIndex", actorIndex).detail("ClientIndex", self->clientId).detail("NumActors", self->actorCountPerClient);
+
+		loop {
+			// used for throttling
+			wait(poisson(&lastTime, delay));
```

Comment:
This poisson distribution is already implemented by fdb. 
The second parameter is used to determine the expected waited time.